crates/sqlexec/src/background_jobs.rs

```diff
@@ -15,33 +9,60 @@ use tokio::{
     task::JoinHandle,
     time::Instant,
 };
-use tracing::{debug, error, info};
+use tracing::{debug, error, info, warn};
 
 use crate::errors::{ExecError, Result};
 
+#[derive(Debug)]
+pub struct JobRunnerOpts {
+    pub health_check_interval: Duration,
+}
+
+impl Default for JobRunnerOpts {
+    fn default() -> Self {
+        Self {
+            // Health check every 2 minutes.
+            health_check_interval: Duration::from_secs(2 * 60),
+        }
+    }
+}
+
 /// Collection of background jobs.
+#[derive(Debug, Clone)]
 pub struct JobRunner {
     sender: mpsc::UnboundedSender<RequestMessage>,
     listen: Arc<Mutex<Option<JoinHandle<()>>>>,
 }
 
-struct JobHandle {
-    join: JoinHandle<()>,
-    sleep: JoinHandle<()>,
-}
-
 #[derive(Debug)]
 enum RequestMessage {
-    NewJob(Box<dyn BgJob>),
+    NewJob(Arc<dyn BgJob>),
     JobComplete(String, Result<()>),
     HealthCheck,
     Close,
 }
 
 impl JobRunner {
     /// Create a new collection of background jobs.
-    pub fn new() -> Self {
+    pub fn new(opts: JobRunnerOpts) -> Self {
         let (sender, receiver) = mpsc::unbounded_channel();
+
+        // Sends a "health check" mesage to the channel.
+        let sender_clone = sender.clone();
+        tokio::spawn(async move {
+            let sender = sender_clone;
+
+            std::panic::set_hook(Box::new(|_| {
+                warn!("health checker exited");
+            }));
```

Comment:
This will replace the existing (global) hook. I think we should remove this and instead just log in the loop below if we get an error.

Reply:
Updated!